controllers/prep_handlers.go

```diff
@@ -342,29 +303,140 @@ func (r *ImageBasedUpgradeReconciler) SetupStateroot(ctx context.Context, ibu *l
 	return nil
 }
 
-func (r *ImageBasedUpgradeReconciler) launchSetupStateroot(ctx context.Context,
-	ibu *lcav1alpha1.ImageBasedUpgrade, progressfile string) (ctrl.Result, error) {
-	result := requeueImmediately()
-	if err := updateProgressFile(progressfile, "started-stateroot"); err != nil {
-		r.Log.Error(err, "failed to update progress file")
-		return result, err
-	}
-	if err := r.SetupStateroot(ctx, ibu); err != nil {
-		r.Log.Error(err, "failed to setup stateroot")
-		if err := updateProgressFile(progressfile, "Failed"); err != nil {
-			r.Log.Error(err, "failed to update progress file")
-			return result, err
+func setPrepFailedStatus(ibu *lcav1alpha1.ImageBasedUpgrade) {
```

Comment:
We have similiar funcs for upgrade status update in controllers/utils/conditions.go. Do we want to move these to that file to maintain them in one place?